helix-core/src/main/java/org/apache/helix/monitoring/metrics/MetricValueType.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.helix.monitoring.metrics;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Represents type of mbean metric's value.
+ * DOUBLE: metric value type is double.
+ * LONG: metric value type is long.
+ */
+public enum MetricValueType {
+  DOUBLE, LONG
```

Comment:
What's the difference between this type andy the original Double and Long?

Reply:
Why I want to have this enum type is, I want to avoid using `InstanceOf` or `typeOf` to check Double or Long types. This enum makes it easier to check Double or Long without impacting the performance like InstanceOf.

Reply:
How about:
ENABLED_STATUS_GAUGE("Enabled", Long.class)
And you don't even need that switch-case block, instead just call the default constructor of the class to create the initial value.